js/ui/date_box/ui.date_box.base.js

```diff
@@ -445,6 +447,36 @@ var DateBox = DropDownEditor.inherit({
         return width;
     },
 
+    _formatValidationIcon: function(longestValue) {
+        const $input = this._input();
```

Comment:
Do not mix es6 and es5, choose one of them

Reply:
I tested it with custom buttons from this demo: https://demos.devexpress.com/ASPNetCore/Demo/Common/CustomTextEditorButtons/. It works.